charon/src/driver.rs

```diff
@@ -230,6 +230,13 @@ pub fn translate(
     // we simply apply some micro-passes to make the code cleaner, before
     // serializing the result.
 
+    // # Micro-pass: Remove overflow/div-by-zero/bounds checks since they are already part of the
+    // arithmetic/array operation in the semantics of (U)LLBC.
+    // **WARNING**: this pass uses the fact that the dynamic checks introduced by Rustc use a
+    // special "assert" construct. Because of this, it must happen *before* the
+    // [reconstruct_asserts] pass. See the comments in [crate::remove_dynamic_checks].
```

Comment:
Also explain that the pass must happen before `simplify_constants` because the latter introduces auxiliary let-bindings which mess up with the patterns we're looking for here.